corehq/apps/analytics/utils.py

```diff
@@ -195,34 +193,68 @@ def _get_contact_ids_to_delete(list_of_emails, retry_num=0):
                 return []
             req.raise_for_status()
         except (ConnectionError, requests.exceptions.HTTPError) as e:
-            metrics_counter(
-                'commcare.hubspot_data.retry.get_contact_ids_for_emails'
-            )
+            if record_metrics:
+                metrics_counter(
+                    'commcare.hubspot_data.retry.get_contact_ids_for_emails'
+                )
             if retry_num <= MAX_API_RETRIES:
-                return _get_contact_ids_to_delete(list_of_emails, retry_num + 1)
-            else:
+                return _get_contacts_from_hubspot(list_of_emails, retry_num + 1)
+            elif record_metrics:
                 metrics_counter(
                     'commcare.hubspot_data.error.get_contact_ids_for_emails',
                     tags={
                         'error': str(e),
                     }
                 )
         else:
-            ids_to_delete = []
-            for contact_id, data in req.json().items():
-                first_conversion_status = data.get(
-                    'properties', {}
-                ).get('first_conversion_clustered_', {}).get('value')
-
-                # If a user's first conversion IS in the allowed list, it means
-                # they have directly interacted with us and we want to keep them
-                # in the email list. However, we will not send project data
-                # about them to HubSpot as they will still be blocked from
-                # updates in track_periodic_data.
-                if first_conversion_status not in ALLOWED_CONVERSIONS:
-                    ids_to_delete.append(contact_id)
-            return ids_to_delete
-    return []
+            return req.json()
+    return {}
+
+
+def get_first_conversion_status_for_emails(list_of_emails):
```

Comment:
Nit: would be nice to move this above `_get_contacts_from_hubspot`, following the [newspaper metaphor](https://github.com/jbarroso/clean-code#the-newspaper-metaphor), since this is the higher-level function and that is an implementation detail.

Reply:
done